src/Compilers/CSharp/Portable/Operations/CSharpConversionExpression.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using Microsoft.CodeAnalysis.Semantics;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    internal abstract class BaseCSharpConversionExpression : BaseConversionExpression
+    {
+        protected BaseCSharpConversionExpression(Conversion conversion, bool isExplicitInCode, bool throwsExceptionOnFailure, bool isChecked, SyntaxNode syntax, ITypeSymbol type, Optional<object> constantValue) :
+            base(conversion, isExplicitInCode, throwsExceptionOnFailure, isChecked, syntax, type, constantValue)
+        {
+            ConversionInternal = conversion;
+        }
+
+        internal Conversion ConversionInternal { get; }
```

Comment:
If creating the `struct` is sufficiently lightweight, then it's reasonable to create it when requested. But either way, please use an explicit method rather than a conversion operator.